test/test_framework.py

```diff
@@ -718,6 +729,13 @@ def test_helper_properties(self):
         self.assertEqual(my_obj.meta, framework.meta)
         self.assertEqual(my_obj.charm_dir, framework.charm_dir)
 
+    def test_ban_concurrent_frameworks(self):
+        framework = self.create_framework()
+        with self.assertRaises(Exception) as cm:
+            framework_copy = self.create_framework()
+            self.fail(f'frameworks {framework} and {framework_copy} got instantiated successfully')
```

Comment:
As a side note, this line doesn't seem to be adding any useful insight. It be saying something about what error condition was expected, but instead it's literally saying that the line above worked fine, which is what assertRaises will already do by default.

Reply:
https://github.com/canonical/operator/pull/141/commits/62c919c9f9687de0b4cd3391bcbef482a511d9a5